exp-player/addon/components/exp-exit-survey/component.js

```diff
@@ -121,23 +122,44 @@ export default ExpFrameBaseComponent.extend(Validations, FullScreen, {
             this.send('next');
         }
     },
-    currentSessionsCompleted: Ember.computed('frameContext', function() {
-        var pastSessions = this.get('frameContext.pastSessions');
-        if (pastSessions) {
-            return pastSessions.get('length') || 1;
+    currentSessionStatus: Ember.computed('frameContext', function() {
+        // Count all sessions
+        //var pastSessions = this.get('frameContext.pastSessions');
+        //if (pastSessions) {
+        //    return pastSessions.get('length') || 1;
+        //}
+
+        // Count only sessions with at least minVideos pref-phys-videos frames
+        var expData = this.get('frameContext.pastSessions').map((session) => session.get('expData'));
+        var expDates = this.get('frameContext.pastSessions').map((session) => moment(session.get('createdOn')));
+        var nSessions = 0;
+        var sessionDates = [];
+        for (var iSess = 0; iSess < expData.length; iSess++) {
+            var keys = Object.keys(expData[iSess]);
+            var nVideos = 0;
+            for (let i = 0; i < keys.length; i++) {
+                var frameKeyName = keys[i];
+                if (frameKeyName.indexOf('pref-phys-videos') !== -1) {
+                    nVideos++;
+                }
+            }
+            if (nVideos >= this.get('minVideosToCountSession')) {
+                nSessions++;
+                sessionDates.push(expDates[iSess]);
+            }
         }
-        return 1;
+        var minDate = moment.min(sessionDates);
+        var maxDate = moment.max(sessionDates);
+        return [nSessions, maxDate.diff(minDate, 'days') + 1];
+    }),
+    currentSessionsCompleted: Ember.computed('currentSessionStatus', function()    {
+        return this.get('currentSessionStatus')[0];
     }),
-    currentDaysSessionsCompleted: Ember.computed('frameContext', function() {
-        // Warning, this implementation may be inaccurate
-        // TODO, figure out what the client's expected behavior is here and resolve
-        // https://openscience.atlassian.net/browse/LEI-111
-        var pastSessionDates = this.get('frameContext.pastSessions').map((session) => moment(session.get('createdOn')));
-        var minDate = moment.min(pastSessionDates);
-        var maxDate = moment.max(pastSessionDates);
 
-        return maxDate.diff(minDate, 'days') + 1;
+    currentDaysSessionsCompleted: Ember.computed('currentSessionStatus', function() {
+        return this.get('currentSessionStatus')[1];
```

Comment:
Instead of accessing first item in array, access appropriate dictionary item:
```
return this.get('currentSessionStatus.daysSessionsCompleted');
```